Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -1171,6 +1171,17 @@ ExceptionOr<void> ContainerNode::append(FixedVector<NodeOrString>&& vector)
         return result.releaseException();
 
     auto newChildren = result.releaseReturnValue();
+    // If the newChildren set has duplicate values, keep only the last occurrence of each element before making them children of this node.
+    for (size_t i = 0; i < newChildren.size(); ++i) {
+        for (size_t j = i+1; j < newChildren.size(); ++j) {
```

Comment:
This is O(n^2) always. That's not great. I think it's better to change the behavior or add a new mode of
`insertChildrenBeforeWithoutPreInsertionValidityCheck` to insert the node that was already inserted
in an earlier iteration. In fact, this behavior must be observable via `MutationObserver` so we should
add a test for that. (that append will insert the same node multiple times into the same parent).

If we're changing the behavior of `ContainerNode::insertChildrenBeforeWithoutPreInsertionValidityCheck`,
this is probably as simple as removing this if statement:
```
        if (child->parentNode()) // Event listeners inserted this child elsewhere.
            break;
```